apps/laboratory/tests/siwe-email.spec.ts

```diff
@@ -0,0 +1,64 @@
+import { testMEmailSiwe } from './shared/fixtures/w3m-email-fixture'
+import type { ModalWalletPage } from './shared/pages/ModalWalletPage'
+import { ModalWalletValidator } from './shared/validators/ModalWalletValidator'
+
+testMEmailSiwe.beforeEach(async ({ modalValidator }) => {
+  const modalWaletValidator = modalValidator as ModalWalletValidator
+  await modalWaletValidator.expectConnected()
+})
+
+testMEmailSiwe('it should sign with email', async ({ modalPage, modalValidator }) => {
+  const modalWaletValidator = modalValidator as ModalWalletValidator
+  await modalPage.sign()
+  await modalPage.approveSign()
+  await modalWaletValidator.expectAcceptedSign()
+})
+
+testMEmailSiwe('it should reject sign in with email', async ({ modalPage, modalValidator }) => {
+  const modalWaletValidator = modalValidator as ModalWalletValidator
+  await modalPage.sign()
+  await modalPage.rejectSign()
+  await modalWaletValidator.expectRejectedSign()
+})
+
+testMEmailSiwe('it should switch network and sign', async ({ modalPage, modalValidator }) => {
+  const modalWalletValidator = modalValidator as ModalWalletValidator
+  const modalWalletPage = modalPage as ModalWalletPage
+  let targetChain = 'Polygon'
+  await modalWalletPage.openAccount()
+  await modalWalletPage.openSettings()
+  await modalWalletPage.switchNetwork(targetChain)
+  await modalWalletPage.page.waitForTimeout(1000)
+  await modalWalletPage.promptSiwe()
+  await modalWalletPage.approveSign()
+  await modalWalletValidator.expectSwitchedNetwork(targetChain)
+  await modalWalletPage.closeModal()
+
+  await modalWalletPage.sign()
+  await modalWalletPage.approveSign()
+  await modalWalletValidator.expectAcceptedSign()
+
+  targetChain = 'Ethereum'
+  await modalWalletPage.openAccount()
+  await modalWalletPage.openSettings()
+  await modalWalletPage.switchNetwork(targetChain)
+  await modalWalletPage.page.waitForTimeout(1000)
```

Comment:
Do we have to wait here? Doesn't it waiting util chain name to be on the screen by itself?

Reply:
Siwe Popup is async, it reacts to state changes on the account so it has a small delay. 
Will test it out as I've changed some things that might make it work regardless 🖖